### ISSUE TITLE
helm does not install dependencies if version is not explicit

### ISSUE THREAD

**Issue by hoesler** [kubernetes/helm#2732]
I am using helm version "2.5.0".

To demonstrate this issue, I created a new chart from scratch with `helm create foo`, removed all templates and values and added mariadb to `requirements.yaml` with version "*" (also tried it with "~0.6" and ">=0.6.0"):

```
dependencies:
  - name: mariadb
    repository: "alias:stable"
    version: "*"
```

Afterwards I ran `helm dep up foo` which correctly fetched `mariadb:0.6.3` into the charts folder.

Now, if I run `helm install --dry-run --debug foo`, I get the following output:
```
[debug] Created tunnel using local port: '60990'

[debug] SERVER: "localhost:60990"

[debug] Original chart version: ""
[debug] CHART PATH: /Users/choesler/helm-test/foo

NAME:   oppulent-hare
REVISION: 1
RELEASED: Thu Jul 27 15:51:57 2017
CHART: foo-0.1.0
USER-SUPPLIED VALUES:
{}

COMPUTED VALUES:
{}

HOOKS:
MANIFEST:
```

All mariadb related resources and values are missing.

If I change the version field to `version: "0.6.3"`, however, and run the same command again (without running `helm dep up` again), helm resolves the dependencies as expected:
```
[debug] Created tunnel using local port: '61005'

[debug] SERVER: "localhost:61005"

[debug] Original chart version: ""
[debug] CHART PATH: /Users/choesler/helm-test/foo

NAME:   youngling-narwhal
REVISION: 1
RELEASED: Thu Jul 27 15:53:44 2017
CHART: foo-0.1.0
USER-SUPPLIED VALUES:
{}

COMPUTED VALUES:
mariadb:
  global: {}
  image: bitnami/mariadb:10.1.23-r2
  imagePullPolicy: IfNotPresent
  persistence:
    accessMode: ReadWriteOnce
    enabled: true
    size: 8Gi
  resources:
    requests:
      cpu: 250m
      memory: 256Mi
  serviceType: ClusterIP

HOOKS:
MANIFEST:

---
# Source: foo/charts/mariadb/templates/secrets.yaml
apiVersion: v1
kind: Secret
metadata:
  name: youngling-narwhal-mariadb
  labels:
    app: youngling-narwhal-mariadb
    chart: "mariadb-0.6.3"
    release: "youngling-narwhal"
    heritage: "Tiller"
type: Opaque
data:
  mariadb-root-password: ""
  mariadb-password: ""
---

##### TRUNCATED #########

```

**Comment by jascott1** [comment]
@hoesler Would it be possible to update to 2.5.1 and re-test? I tried to reproduce this on 2.5.1 but could not.

**Comment by hoesler** [comment]
You are right, it seems to be fixed in 2.5.1.

**Comment by jascott1** [comment]
@hoesler Ok I will close this issue but please let me know if the problem reappears.
